src/generate-oas.ts

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import path from "path";
+import fs from "fs-extra";
+import { generateFromOas, download } from "@commerce-apps/raml-toolkit";
+import Handlebars from "handlebars";
+
+type ApiSpecDetail = {
+  filepath: string;
+  filename: string;
+  name: string;
+  apiName: string;
+  directoryName: string;
+};
+
+const API_DIRECTORY = path.resolve(`${__dirname}/../apis`);
+const STATIC_DIRECTORY = path.join(__dirname, "./static");
+const TARGET_DIRECTORY = path.resolve(`${__dirname}/../renderedTemplates`);
+const TEMPLATE_DIRECTORY = path.resolve(`${__dirname}/../templatesOas`);
+const INDEX_TEMPLATE_LOCATION = path.resolve(
+  `${__dirname}/../templates/index.ts.hbs`
+);
+const VERSION_TEMPLATE_LOCATION = path.resolve(
+  `${__dirname}/../templates/version.ts.hbs`
+);
+
+function kebabToCamelCase(str: string): string {
+  return str.replace(/-([a-z])/g, (match, letter: string) =>
+    letter.toUpperCase()
+  );
+}
+
+/**
+ * Helper function. Also contains explicit workaround for some API names.
+ */
+export function resolveApiName(name: string): string {
+  if (name === "Shopper orders OAS") {
+    return "ShopperOrders";
+  }
+  if (name === "CDN API - Process APIs OAS") {
+    return "CDNZones";
+  }
+  if (name === "SLAS Admin-UAP OAS") {
+    return "SlasAdmin";
+  }
+  if (name === "Shopper Seo OAS") {
+    return "ShopperSEO";
+  }
+  if (name === "Shopper Context OAS") {
+    return "ShopperContexts";
+  }
+  if (name === "Catalogs OAS") {
+    return "CatalogsAPI";
+  }
+  if (name === "Cors Preferences OAS") {
+    return "CORSPreferences";
+  }
+  return name.replace(/\s+/g, "").replace("OAS", "");
+}
+
+/**
+ * Helper that reads api details from exchange.json
+ */
+export function getAPIDetailsFromExchange(directory: string): ApiSpecDetail {
+  const exchangePath = path.join(directory, "exchange.json");
+  if (fs.existsSync(exchangePath)) {
+    const exchangeConfig = fs.readJSONSync(
+      exchangePath
+    ) as download.ExchangeConfig;
+
+    // Special handling for shopper-baskets-v2
+    if (
+      exchangeConfig.assetId === "shopper-baskets-oas" &&
+      exchangeConfig.apiVersion === "v2"
+    ) {
```

Comment:
ShopperBasketsV2 is not generating for me... but maybe I don't have the right changes in RAML toolkit, I'm pulling from `main` after Ujwala's branch got merged in yesterday, are you able to generate shopper baskets v1/v2 without manual intervention?

Reply:
Yes, I am able to generate ShopperBasketsV2. The only manual intervention I have had to do is fixing calls to `StaticClient` after the API has been generated.